drivers/include/lis3dh.h

```diff
@@ -792,51 +794,64 @@ int lis3dh_set_aux_adc(lis3dh_t *dev, const uint8_t enable, const uint8_t temper
 int lis3dh_set_axes(lis3dh_t *dev, const uint8_t axes);
 
 /**
- * @brief Set the FIFO mode.
+ * @brief Enable/disable the FIFO.
  *
  * @param[in]  dev          Device descriptor of sensor
- * @param[in]  mode         The chosen FIFO mode.
+ * @param[in]  mode         FIFO mode, see data sheet for details.
+ * @param[in]  watermark    Watermark level for FIFO level interrupts
  *
  * @return                  0 on success
  * @return                  -1 on error
  */
-int lis3dh_set_fifo_mode(lis3dh_t *dev, const lis3dh_fifo_mode_t mode);
+int lis3dh_set_fifo(lis3dh_t *dev, const uint8_t mode, const uint8_t watermark);
 
 /**
- * @brief Enable/disable the FIFO.
+ * Set the output data rate of the sensor.
  *
  * @param[in]  dev          Device descriptor of sensor
- * @param[in]  enable       If zero, disable the FIFO, otherwise enables the FIFO.
+ * @param[in]  odr          Chosen output data rate.
  *
  * @return                  0 on success
  * @return                  -1 on error
  */
-int lis3dh_set_fifo(lis3dh_t *dev, const uint8_t enable);
+int lis3dh_set_odr(lis3dh_t *dev, const uint8_t odr);
 
 /**
- * Set the output data rate of the sensor.
+ * @brief Set the full scale range of the sensor.
+ *
+ * Valid values for scale is 2, 4, 8, 16 and represents the full range of the
+ * sensor.
  *
  * @param[in]  dev          Device descriptor of sensor
- * @param[in]  odr          Chosen output data rate.
+ * @param[in]  scale        The chosen sensitivity scale.
  *
  * @return                  0 on success
  * @return                  -1 on error
  */
-int lis3dh_set_odr(lis3dh_t *dev, const lis3dh_odr_t odr);
+int lis3dh_set_scale(lis3dh_t *dev, const uint8_t scale);
 
 /**
- * @brief Set the full scale range of the sensor.
+ * @brief Set INT1 pin function
  *
- * Valid values for scale is 2, 4, 8, 16 and represents the full range of the
- * sensor.
+ * Set the bits of CTRL_REG3 for choosing sources for the INT1 pin.
  *
  * @param[in]  dev          Device descriptor of sensor
- * @param  scale            The chosen sensitivity scale.
+ * @param[in]  mode         CTRL_REG3 value, see data sheet for details.
  *
  * @return                  0 on success
  * @return                  -1 on error
  */
-int lis3dh_set_scale(lis3dh_t *dev, const lis3dh_scale_t scale);
+int lis3dh_set_int1(lis3dh_t *dev, const uint8_t mode);
```

Comment:
INT1 and INT2 are the signal names for the interrupt pins in the data sheet. I did not implement any INT2 functionality, but I think it is more clear if I call the function _int1 instead of _int since someone might want to implement INT2 in the future.